src/node.cc

```diff
@@ -2402,8 +2402,11 @@ void DLOpen(const FunctionCallbackInfo<Value>& args) {
     char errmsg[1024];
     snprintf(errmsg,
              sizeof(errmsg),
-             "Module version mismatch. Expected %d, got %d.",
-             NODE_MODULE_VERSION, mp->nm_version);
+             "This module was compiled against a different Node.js version "
+             "using NODE_MODULE_VERSION %d. This version of Node.js requires "
+             "NODE_MODULE_VERSION %d. Please try re-compiling or re-installing "
+             "the module (for instance, using `npm install` or `npm rebuild`).",
```

Comment:
I would use the same order than the line before: npm rebuild corresponds to re-compiling and npm install to re-installing.